### PR TITLE
feat: Added 401 fallback to re-generate access_token

### PR DESCRIPTION
**Pull request by ImBenHayward**

### CODE REVIEW

File: src/factories/request.js [hunk 1]
```diff
@@ -57,10 +58,31 @@ const fetchRetry = (
   version,
   headers,
   requestBody,
-  attempt = 1
+  attempt = 1,
+  authenticate
 ) =>
   new Promise((resolve, reject) => {
     const ver = version || config.version
+    const request = new RequestFactory(config)
```
Comment (samblacklock) [inline src/factories/request.js:66]: Can remove this now

File: src/factories/request.js [hunk 2]
```diff
@@ -75,23 +97,13 @@ const fetchRetry = (
...
-              )
-                .then(result => resolve(result))
-                .catch(error => reject(error)),
-            attempt * config.retryDelay +
-              Math.floor(Math.random() * config.retryJitter)
-          )
+        if (attempt < config.fetchMaxAttempts) {
+          if (response.status === 401) {
+            console.log('retry')
+            this.authenticate().then(retryTimeout())
```
Comment (samblacklock) [inline src/factories/request.js:103]:
```suggestion
           authenticate().then(retryTimeout())
```

doesn't need the this chaining here

File: examples/index.html
Comment (field123) [inline examples/index.html:1]: This file shouldn't be changed, only locally for testing

File: src/factories/request.js [hunk 3]
```diff
@@ -75,23 +100,12 @@ const fetchRetry = (
...
-                headers,
-                requestBody,
-                attempt + 1
-              )
-                .then(result => resolve(result))
-                .catch(error => reject(error)),
-            attempt * config.retryDelay +
-              Math.floor(Math.random() * config.retryJitter)
-          )
+        if (attempt < config.fetchMaxAttempts) {
```
Comment (field123) [inline src/factories/request.js:103]: would be good to get a unit test to check this logic works as expected

**Review by field123 — DISMISSED** [review]

**Review by samblacklock — APPROVED** [review]

**Pull request merged**